firmware/library/L1_Drivers/onboardled.hpp

```diff
@@ -76,6 +101,8 @@ class OnBoardLed: public OnBoardLedInterface
   }
 
  protected:
-  uint8_t led_states;
-  Gpio LED0(main_port_number, 1);
+  Gpio LED0(main_port_number, led0_port_specifier);
+  Gpio LED1(main_port_number, led1_port_specifier);
+  Gpio LED2(main_port_number, led2_port_specifier);
+  Gpio LED3(main_port_number, led3_port_specifier);
```

Comment:
This could be an array of Gpio objects. No need to use a constexpr variable for the port and pin, you can put them in directly. This is because, the numbers aren't magic if it is obvious what they are.